src/Sarif.Viewer.VisualStudio/OpenLogFileCommands.cs

```diff
@@ -168,117 +163,80 @@ private async System.Threading.Tasks.Task MenuItemCallbackAsync(object sender, E
-                    {
-                        toolFormat = ToolFormat.Fortify;
-                        title = "Open Fortify XML log file";
-                        filter = "Fortify log files (*.xml)|*.xml";
-                        break;
-                    }
-                    case OpenFortifyFprFileCommandId:
-                    {
-                        toolFormat = ToolFormat.FortifyFpr;
-                        title = "Open Fortify FPR log file";
-                        filter = "Fortify FPR log files (*.fpr)|*.fpr";
-                        break;
-                    }
-                    case OpenCppCheckFileCommandId:
-                    {
-                        toolFormat = ToolFormat.CppCheck;
-                        title = "Open CppCheck XML log file";
-                        filter = "CppCheck log files (*.xml)|*.xml";
-                        break;
-                    }
-                    case OpenClangFileCommandId:
-                    {
-                        toolFormat = ToolFormat.ClangAnalyzer;
-                        title = "Open Clang XML log file";
-                        filter = "Clang log files (*.xml)|*.xml";
-                        break;
-                    }
-                    case OpenAndroidStudioFileCommandId:
-                    {
-                        toolFormat = ToolFormat.AndroidStudio;
-                        title = "Open Android Studio XML log file";
-                        filter = "Android Studio log files (*.xml)|*.xml";
-                        break;
-                    }
-                    case OpenSemmleFileCommandId:
-                    {
-                        toolFormat = ToolFormat.SemmleQL;
-                        title = "Open Semmle QL CSV log file";
-                        filter = "Semmle QL log files (*.csv)|*.csv";
-                        break;
-                    }
-                    case OpenPylintFileCommand:
-                    {
-                        toolFormat = ToolFormat.Pylint;
-                        title = "Open Pylint JSON log file";
-                        filter = "Pylint log files (*.json)|*.json";
-                        break;
-                    }
-                    case OpenTSLintFileCommand:
-                    {
-                        toolFormat = ToolFormat.TSLint;
-                        title = "Open TSLint JSON log file";
-                        filter = "TSLint log files (*.json)|*.json";
-                        break;
-                    }
-                }
+                // Need two additional entries for "SARIF" which is at the beginning of the list.
+                List<string> toolFormatFilters = new List<string>(toolFormatFieldInfos.Length + 1);
 
-                filter += "|All files (*.*)|*.*";
+                // The reflected enumeration of the fields in tool format is not necessarily in
+                // the same order it is presented in the class (as typed by the developer).
```

Comment:
>developer [](start = 80, length = 9)

The order of the enumeration constants in ToolFormat.cs shouldn't affect the UI. IMO we should present them alphabetically:
```
IEnumerable<FieldInfo> toolFormatFieldInfos = typeof(ToolFormat).GetFields()
    .Where(fi => !fi.Name.Equals(nameof(ToolFormat.None), StringComparison.Ordinal))
    .OrderBy(fi => fi.Name);
```